vite plugin configuration docs (#21137)
cloudflare/cloudflare-docs: src/content/docs/workers/wrangler/configuration.mdx

  Platform products.
---

> import { Render, Type, MetaInfo, WranglerConfig, Details } from "~/components";

Author: ```suggestion
import { Render, Type, MetaInfo, WranglerConfig } from "~/components";
```

- `define` <Type text="Record<string, string>" /> <MetaInfo text="optional" />

  - A map of values to substitute when deploying your Worker.

>   - If you're using the [Cloudflare Vite plugin](/workers/vite-plugin/), this is replaced by Vite's [`define`](https://vite.dev/config/shared-options.html#define).

Reviewer: ```suggestion
  - If you're using the [Cloudflare Vite plugin](/workers/vite-plugin/), `define` is replaced by Vite's [`define`](https://vite.dev/config/shared-options.html#define).
```

## Custom builds

> ::: note

Reviewer 2: ```suggestion
:::note
```

Otherwise we get this: 
![Screenshot 2025-03-26 at 11 40 01](https://github.com/user-attachments/assets/35d95af4-e052-4938-8bae-97cc2d4cd647)

🥲

## Bundling

> ::: note

Reviewer 2: ```suggestion
:::note
```

as above

## Local development settings

> Q: is this replaced by vite's server options?

Reviewer 2: Yes 👍

It is unlikely that an application developer will need to use this feature, but it is documented here to help you understand when Wrangler is using a generated configuration rather than the original, user's configuration.

> For example, when you are using the [Cloudflare Vite plugin](/workers/vite-plugin/), it will generate a Wrangler configuration file when you run `vite build`, which is then used for preview and deployment.

Reviewer: ```suggestion
For example, when using the [Cloudflare Vite plugin](/workers/vite-plugin/), an output Worker configuration file is generated as part of the build. This is then used for preview and deployment.
```
